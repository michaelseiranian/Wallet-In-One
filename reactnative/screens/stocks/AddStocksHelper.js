import { auth_post } from "../../authentication";

/**
 * Custom react hook that is used to connect to the backend when the user has succesfully added an account via the SDK
 */
export default function AddStocksHelper(){

    /**
    * Function that takes in a temporary public token which is then exchanged for an access token for connecting to endpoints
    * @param publicToken
    * @returns access_token
    */
    const getAccessToken = async (publicToken) => {
        const body = {
          public_token: publicToken
        }
        const response = await auth_post('/stocks/get_access_token/', body)
        return response.body.access_token
    }

    /**
     * Function that accesses the balance of as stock account by using the access token generated
     * @param accessToken 
     * @returns balance
     */
    const getBalance = async (accessToken) => {
        const body = {
            access_token: accessToken
        }
        const response = await auth_post('/stocks/get_balance/', body)
        const data = response.body;
        return (parseFloat(data.accounts[0].balances.current)*0.83).toFixed(2)
    }

    /**
     * Function that accesses the transactions for a stock account by using the access token generated
     * @param accessToken 
     * @returns transactions
     */
    const getTransaction = async (accessToken) => {
        const body = {
          access_token: accessToken
        }
        const response = await auth_post('/stocks/get_transactions/', body)
        return response.body
    }

    /**
     * Function that adds a stock account to the backend database
     * @param {*} account 
     * @param {*} success 
     * @param {*} access_token 
     * @param {*} balance 
     * @param {*} image 
     */
    const addAccount = async (account, success, access_token, balance, image) => {
        const account_data = {
          account_id: account._id,
          name: account.meta.name,
          institution_name: success.metadata.institution.name,
          institution_id: success.metadata.institution.id,
          access_token: access_token,
          balance: balance,
          institution_logo: image,
        }
        const response = await auth_post('/stocks/add_stock_account/', account_data);
        return response.status;
    };

    /**
     * Function that adds a transaction to the backend database
     * @param {*} element 
     * @param {*} fetched_transaction_list 
     */
    const addTransaction = async (element, fetched_transaction_list) => {
        let latitude = parseFloat(((Math.random() * (7) + 35.5).toFixed(3))) // Randomly Generate Location since Plaid does not provide transaction location specifically for Investment Transactions
        let longitude = parseFloat(((Math.random() * (43) + 77).toFixed(3))) * -1
        const body = {
          account_id: element.account_id,
          investment_transaction_id: element.investment_transaction_id,
          security_id: element.security_id,
          date: element.date,
          name: element.name,
          quantity: element.quantity,
          amount: element.amount * 0.83, // to convert to GBP
          price: element.price,
          fees: element.fees,
          stock: fetched_transaction_list.accounts[0].account_id,
          latitude: latitude,
          longitude: longitude
        }
        await auth_post('/stocks/add_transaction_account/', body)
    };

    /**
     * Function that adds a Stock to the backend database
     * @param {*} stock 
     * @param {*} stockInfo 
     */
    const addStock = async (stock, stockInfo) => {
        const body = {
          institution_price: (stock.institution_price).toFixed(2),
          quantity: stock.quantity,
          name: stockInfo.name,
          ticker_symbol: stockInfo.ticker_symbol,
          stockAccount: stock.account_id,
          security_id: stockInfo.security_id
        }
        await auth_post('/stocks/add_stock/', body)
    }

    /**
     * Function that accesses the logo of the specified institution
     * @param {*} success 
     * @returns logo
     */
    const getLogo = async (success) => {
        const body = {
          name: success.metadata.institution.name
        }
        const response = await auth_post('/stocks/get_logo/', body)
        return response.body.logo
    }

    /**
     * Function that accesses all stock holdings a specified stock account
     * @param {*} accessToken 
     * @returns stocks
     */
    const getStocks = async (accessToken) => {
        const body = {
          access_token: accessToken
        }
        const response = await auth_post('/stocks/get_stocks/', body)
        const data = response.body;
        return [data.holdings, data.securities]
    }

    return { getAccessToken, getBalance, getTransaction, addAccount, addTransaction, addStock, getLogo, getStocks };
}