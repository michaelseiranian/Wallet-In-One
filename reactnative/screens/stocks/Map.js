import MapView from 'react-native-maps';
import { Marker } from 'react-native-maps';
import { StyleSheet } from 'react-native';

/**
 * Component that takes in a latitude and longitude and displays a map with a marker at the specified location
 */
export default function Map({latitude, longitude}){
  if(latitude && longitude){
    return(
        <MapView
        style={{...StyleSheet.absoluteFillObject}}
        region={{
          latitude: latitude,
          longitude: longitude,
          latitudeDelta: 0.0922,
          longitudeDelta: 0.0421,
        }}
        testID={"mapView"}
      >
    <Marker coordinate={{ latitude: latitude, longitude: longitude }} testID={"marker"} />
      </MapView>
    )
  }
  else{
    return null;
  }
}