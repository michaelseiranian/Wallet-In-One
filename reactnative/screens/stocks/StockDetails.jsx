import { View, Text, TouchableOpacity,StyleSheet, Dimensions } from 'react-native';
import React, { useState, useEffect, useCallback } from 'react';
import { FlatList, ScrollView } from 'react-native-gesture-handler';
import {Table, Row, Rows,TableWrapper,Cell} from 'react-native-table-component';
import { useTheme } from "reactnative/src/theme/ThemeProvider";
import { styles } from "reactnative/screens/All_Styles.style.js";
import LineChartScreen from '../charts/LineChart';
import { auth_get } from '../../authentication';

import { ConvertTransactionsToGraphCompatibleData } from '../helper';

// Define the StockDetails component that is used to display stock information
export default function StockDetails({ route, navigation }){

  const [stockTransactions, setStockTransactions] = useState();
  const stock = route.params.stock;
  const {dark, colors, setScheme } = useTheme();
  let current_balance = null;

  const [transformedData, setTransformedData] = useState();

  const {width: SIZE} = Dimensions.get('window');

  // Define a function to get the stock transactions from the server and update the state variables
  let getStockTransactions = useCallback(async (stock) => {
    try {
      const response = await auth_get(`/stocks/list_transactions/${stock}/`)
      let data = await response.body;

      // Filter the transactions to only include those related to the current stock
      data = data.filter(item => item.security_id === route.params.stock.security_id);
      // Set the state variables for the stock transactions and transformed graph data
      setStockTransactions(data);
      // Calculate the current balance of the stock transactions
      current_balance = data.map(item => item.amount).reduce((acc, current) => acc + current, 0);
      // Transform the transaction data into a format that is compatible with line graphs
      setTransformedData(ConvertTransactionsToGraphCompatibleData(data, current_balance));
    } catch (error) {
    }
  }, []);

  // Define a useEffect hook that is triggered when the stock or getStockTransactions function changes
  useEffect(() => {
    // Call the getStockTransactions function with the current stock if it exists
    if (stock) {
      getStockTransactions(stock.stockAccount);
    }

  }, [stock, getStockTransactions]);

  // Format the transaction data in a table compatible format.
  const data = {
    tableHead: ['ID', 'Amount', 'Date', 'Quantity', 'Fees'],
    tableData: stockTransactions?.map(item => [
      item.id,
      (item.amount).toFixed(2),
      item.date,
      item.quantity,
      item.fees,
    ]),
  };

  const stylesInternal = StyleSheet.create({
    row:{
      flexDirection: 'row',
      alignItems: 'flex-start',
    },
    name:{
      color: "white",
      fontWeight: 'bold',
      fontSize: 21,
    },
    ins_name:{
      color: "white",
      fontSize: 18,
    },
    table: {
      flex: 1,
      justifyContent: 'center',
      backgroundColor: colors.background,
      paddingTop: 20,
    },
    head: {
      height: 44,
       backgroundColor: '#42b983'
    },
    row: { 
      flexDirection: 'row',
    },
  });
    return (
      <ScrollView style={[styles(dark, colors).container]}>
        {stock ? (
          <View style={{padding: 20}}>
            <Text style={[styles(dark, colors).textBold, {color: colors.text}]}>Name</Text>
            <Text style={styles(dark, colors).text}>{stock.name}{"\n"}</Text>

            <Text style={[styles(dark, colors).textBold, {color: colors.text}]}>Institution Price Currency</Text>
            <Text style={styles(dark, colors).text}>{stock.institution_price_currency}{"\n"}</Text>

            <Text style={[styles(dark, colors).textBold, {color: colors.text}]}>Institution Price</Text>
            <Text style={styles(dark, colors).text}>{stock.institution_price}{"\n"}</Text>

            {stock.ticker_symbol != null && (
              <View>
                <Text style={[styles(dark, colors).textBold, {color: colors.text}]}>Ticker Symbol</Text>
                <Text style={styles(dark, colors).text}>{stock.ticker_symbol}{"\n"}</Text>
              </View>
            )}

            <Text style={[styles(dark, colors).textBold, {color: colors.text}]}>Quantity</Text>
            <Text style={styles(dark, colors).text}>{stock.quantity}{"\n"}</Text>

          </View>
        ):(<Text style={styles(dark, colors).text}>Loading...</Text>)}

          <Text style={[styles(dark, colors).textBold, {color: colors.text, fontSize: 21, padding: 20}]}>Line Graph</Text>

          {transformedData && (
            <LineChartScreen
              graph_version={1} 
              height={SIZE / 2} 
              width={SIZE}
              current_balance={current_balance}
              data={transformedData}
            />
          )}

        <Text style={[styles(dark, colors).textBold, {color: colors.text, fontSize: 21, padding: 20}]}>Stock Transactions</Text>
        {data && (
          <View style={[stylesInternal.table], {padding: 20}}>
              {data && data.tableData && data.tableData.length > 0 ? (
                <View>
                  <Table style={{paddingBottom:20}} borderStyle={{ borderWidth: 2, borderColor: '#42b983' }}>
                    <Row data={data.tableHead} style={stylesInternal.head} />
                    {data.tableData.map((rowData, index) => (
                      <TouchableOpacity key={index} onPress={() => navigation.navigate('TransactionData', { id: rowData[0] })}>
                        <TableWrapper style={[stylesInternal.row, {backgroundColor: rowData[1] < 0 ? "#f87171" : '#bbf7d0'}]} borderStyle={{borderWidth: 1, borderColor: '#000000'}}>
                          {rowData.map((cellData, cellIndex) => (
                            <Cell key={cellIndex} data={cellData} textStyle={{textAlign: 'center', fontSize: 12}} />
                          ))}
                        </TableWrapper>
                      </TouchableOpacity>
                    ))}
                  </Table>
                </View>
              ) : (
                <Text style={[styles(dark, colors).text, {textAlign: 'center', alignSelf: 'center'}]}>Loading...</Text>
              )}
          </View>
          )}
      </ScrollView>
  );
    
}