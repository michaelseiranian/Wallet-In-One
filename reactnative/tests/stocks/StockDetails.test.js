import React from 'react';
import { render, screen, fireEvent, act, waitFor} from '@testing-library/react-native';
import StockDetails from '../../screens/stocks/StockDetails';
import transactions from './fixtures/transactions.json';
import stocks from './fixtures/stocks.json'

/**
 * Tests for StockDetails Screen
 */
describe('<StockDetails />', () => {

    beforeEach(() => {
        global.fetch =  jest.fn( async () => {
            return Promise.resolve({
                status: 200, json: () => transactions.transactions
            })
        })
      })

      afterEach(() => {
        global.fetch.mockClear();
        delete global.fetch;
      })

    it('stockDetails snapshot test', async () => {
        const stockDetails = render(<StockDetails route={{params: {"stock": stocks.stock}}}/>);

        await act( async () => {
            await waitFor( () => {
                const loading = screen.queryByText('Loading...');
                expect(loading).toBeNull();
            })
        });
        expect(stockDetails).toMatchSnapshot();
    }
    )

    it('stockDetails name test', async () => {
        const stockDetails = render(<StockDetails route={{params: {"stock": stocks.stock}}}/>);
        await act( () => {
            expect(screen.getByText("Name"))
            expect(screen.getByText("Nflx Feb 01'18 $355 Call"))
        });
    }
    )

    it('stockDetails currency test', async () => {
        const stockDetails = render(<StockDetails route={{params: {"stock": stocks.stock}}}/>);
        await act( () => {
            expect(screen.getByText("Institution Price Currency"))
            expect(screen.getByText("GBP"))
        });
    }
    )

    it('stockDetails price test', async () => {
        const stockDetails = render(<StockDetails route={{params: {"stock": stocks.stock}}}/>);
        await act( () => {
            expect(screen.getByText("Institution Price"))
            expect(screen.getByText("1.00"))
        });
    }
    )

    it('stockDetails ticker symbol test', async () => {
        const stockDetails = render(<StockDetails route={{params: {"stock": stocks.stock}}}/>);
        await act( () => {
            expect(screen.getByText("Ticker Symbol"))
            expect(screen.getByText("NFLX180201C00355000"))
        });
    }
    )

    it('stockDetails quantity test', async () => {
        const stockDetails = render(<StockDetails route={{params: {"stock": stocks.stock}}}/>);
        await act( () => {
            expect(screen.getByText("Quantity"))
            expect(screen.getByText("0.01"))
        });
    }
    )
})