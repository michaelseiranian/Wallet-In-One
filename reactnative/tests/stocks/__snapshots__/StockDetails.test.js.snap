// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StockDetails /> stockDetails snapshot test 1`] = `
<RCTScrollView
  collapsable={false}
  handlerTag={1}
  handlerType="NativeViewGestureHandler"
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    [
      {
        "backgroundColor": "white",
        "flex": 1,
      },
    ]
  }
  waitFor={
    [
      {
        "current": null,
      },
    ]
  }
>
  <View>
    <View
      style={
        {
          "padding": 20,
        }
      }
    >
      <Text
        style={
          [
            {
              "color": "black",
              "fontWeight": "700",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Name
      </Text>
      <Text
        style={
          {
            "color": "black",
          }
        }
      >
        Nflx Feb 01'18 $355 Call
        

      </Text>
      <Text
        style={
          [
            {
              "color": "black",
              "fontWeight": "700",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Institution Price Currency
      </Text>
      <Text
        style={
          {
            "color": "black",
          }
        }
      >
        GBP
        

      </Text>
      <Text
        style={
          [
            {
              "color": "black",
              "fontWeight": "700",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Institution Price
      </Text>
      <Text
        style={
          {
            "color": "black",
          }
        }
      >
        1.00
        

      </Text>
      <View>
        <Text
          style={
            [
              {
                "color": "black",
                "fontWeight": "700",
              },
              {
                "color": "black",
              },
            ]
          }
        >
          Ticker Symbol
        </Text>
        <Text
          style={
            {
              "color": "black",
            }
          }
        >
          NFLX180201C00355000
          

        </Text>
      </View>
      <Text
        style={
          [
            {
              "color": "black",
              "fontWeight": "700",
            },
            {
              "color": "black",
            },
          ]
        }
      >
        Quantity
      </Text>
      <Text
        style={
          {
            "color": "black",
          }
        }
      >
        0.01
        

      </Text>
    </View>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
            "fontSize": 21,
            "padding": 20,
          },
        ]
      }
    >
      Line Graph
    </Text>
    <View>
      <View
        style={
          {
            "flexDirection": "row",
            "paddingBottom": 14,
            "paddingHorizontal": 10,
          }
        }
      >
        <Text
          style={
            {
              "color": "green",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
          -38.446
        </Text>
        <Text
          style={
            {
              "color": "green",
              "fontSize": 14,
              "fontWeight": "bold",
            }
          }
        >
           (
          -100.00
          %)
        </Text>
      </View>
      <View
        style={
          [
            undefined,
            undefined,
          ]
        }
      >
        <View
          style={
            {
              "height": 375,
              "width": 750,
            }
          }
        >
          <RNSVGSvgView
            bbHeight="375"
            bbWidth="750"
            focusable={false}
            height={375}
            style={
              [
                {
                  "backgroundColor": "transparent",
                  "borderWidth": 0,
                },
                {
                  "flex": 0,
                  "height": 375,
                  "width": 750,
                },
              ]
            }
            width={750}
          >
            <RNSVGGroup>
              <RNSVGPath
                fill={
                  {
                    "payload": 0,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeOpacity",
                  ]
                }
                stroke={
                  {
                    "payload": 4278222848,
                    "type": 0,
                  }
                }
                strokeOpacity={0.2}
                strokeWidth="3"
              />
              <RNSVGDefs>
                <RNSVGLinearGradient
                  gradient={
                    [
                      0.2,
                      637566976,
                      0.4,
                      218136576,
                      1,
                      32768,
                    ]
                  }
                  gradientTransform={null}
                  gradientUnits={0}
                  name="1"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="100%"
                />
              </RNSVGDefs>
              <RNSVGPath
                fill={
                  {
                    "brushRef": "1",
                    "type": 1,
                  }
                }
                propList={
                  [
                    "fill",
                  ]
                }
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <View
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNSVGSvgView
            animatedProps={
              {
                "width": 770,
              }
            }
            bbHeight="375"
            focusable={false}
            height={375}
            style={
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              }
            }
          >
            <RNSVGGroup>
              <RNSVGPath
                fill={
                  {
                    "payload": 0,
                    "type": 0,
                  }
                }
                propList={
                  [
                    "fill",
                    "stroke",
                    "strokeWidth",
                    "strokeOpacity",
                  ]
                }
                stroke={
                  {
                    "payload": 4278222848,
                    "type": 0,
                  }
                }
                strokeOpacity={1}
                strokeWidth="3"
              />
            </RNSVGGroup>
          </RNSVGSvgView>
        </View>
        <RNSVGLine
          propList={
            [
              "stroke",
              "strokeWidth",
              "strokeDasharray",
            ]
          }
          stroke={
            {
              "payload": 4286611584,
              "type": 0,
            }
          }
          strokeDasharray={
            [
              "3",
              "3",
            ]
          }
          strokeWidth="2"
          x1="0"
          x2="0"
          y1="0"
          y2="0"
        />
        <View
          collapsable={false}
          handlerTag={2}
          handlerType="LongPressGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              {
                "height": "100%",
                "opacity": 0,
                "transform": [
                  {
                    "translateX": -1,
                  },
                ],
              }
            }
          >
            <RNSVGSvgView
              bbHeight="100%"
              focusable={false}
              style={
                [
                  {
                    "backgroundColor": "transparent",
                    "borderWidth": 0,
                  },
                  {
                    "bottom": 0,
                    "height": "100%",
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                ]
              }
            >
              <RNSVGGroup>
                <RNSVGLine
                  propList={
                    [
                      "stroke",
                      "strokeWidth",
                      "strokeDasharray",
                    ]
                  }
                  stroke={
                    {
                      "payload": 4286611584,
                      "type": 0,
                    }
                  }
                  strokeDasharray={
                    [
                      "3",
                      "3",
                    ]
                  }
                  strokeWidth="2"
                  x1="0"
                  x2="0"
                  y1="0"
                  y2="375"
                />
              </RNSVGGroup>
            </RNSVGSvgView>
          </View>
        </View>
        <View
          collapsable={false}
          handlerTag={3}
          handlerType="LongPressGestureHandler"
          onGestureHandlerEvent={[Function]}
          onGestureHandlerStateChange={[Function]}
          style={
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            style={
              [
                {
                  "alignItems": "center",
                  "height": 32,
                  "justifyContent": "center",
                  "width": 32,
                },
                {
                  "transform": [
                    {
                      "translateX": -17,
                    },
                    {
                      "translateY": -16,
                    },
                    {
                      "scale": 0,
                    },
                  ],
                },
                undefined,
              ]
            }
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "black",
                    "borderRadius": 32,
                    "height": 32,
                    "opacity": 0.1,
                    "position": "absolute",
                    "width": 32,
                  },
                  undefined,
                ]
              }
            />
            <View
              style={
                [
                  {
                    "backgroundColor": "black",
                    "borderRadius": 8,
                    "height": 8,
                    "width": 8,
                  },
                  undefined,
                ]
              }
            />
          </View>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontSize": 12,
                "marginHorizontal": 5,
              }
            }
          >
            Date: 
          </Text>
          <TextInput
            animatedProps={
              {
                "text": "",
              }
            }
            editable={false}
            style={
              [
                {
                  "color": "black",
                },
                {
                  "color": "black",
                  "fontSize": 13,
                  "fontWeight": "bold",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              {
                "color": "black",
                "fontSize": 12,
                "marginHorizontal": 5,
              }
            }
          >
            Balance: 
          </Text>
          <TextInput
            animatedProps={
              {
                "text": "",
              }
            }
            editable={false}
            style={
              [
                {
                  "color": "black",
                },
                {
                  "color": "black",
                  "fontSize": 13,
                  "fontWeight": "bold",
                },
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
      </View>
    </View>
    <Text
      style={
        [
          {
            "color": "black",
            "fontWeight": "700",
          },
          {
            "color": "black",
            "fontSize": 21,
            "padding": 20,
          },
        ]
      }
    >
      Stock Transactions
    </Text>
    <View
      style={
        {
          "padding": 20,
        }
      }
    >
      <View>
        <View
          style={
            [
              {
                "paddingBottom": 20,
              },
              {
                "borderBottomWidth": 2,
                "borderColor": "#42b983",
                "borderLeftWidth": 2,
              },
            ]
          }
        >
          <View
            style={
              [
                undefined,
                0,
                {
                  "flexDirection": "row",
                  "overflow": "hidden",
                },
                {
                  "backgroundColor": "#42b983",
                  "height": 44,
                },
              ]
            }
          >
            <View
              style={
                [
                  {
                    "borderColor": "#42b983",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      undefined,
                      undefined,
                    ],
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                ID
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "borderColor": "#42b983",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      undefined,
                      undefined,
                    ],
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                Amount
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "borderColor": "#42b983",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      undefined,
                      undefined,
                    ],
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                Date
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "borderColor": "#42b983",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      undefined,
                      undefined,
                    ],
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                Quantity
              </Text>
            </View>
            <View
              style={
                [
                  {
                    "borderColor": "#42b983",
                    "borderRightWidth": 2,
                    "borderTopWidth": 2,
                  },
                  {
                    "justifyContent": "center",
                  },
                  undefined,
                  undefined,
                  undefined,
                  {
                    "flex": 1,
                  },
                  undefined,
                ]
              }
            >
              <Text
                style={
                  [
                    [
                      undefined,
                      undefined,
                    ],
                    {
                      "backgroundColor": "transparent",
                    },
                  ]
                }
              >
                Fees
              </Text>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "backgroundColor": "#bbf7d0",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  1001
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  38.45
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  2023-03-10
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  4211.152345617756
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  5
                </Text>
              </View>
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "backgroundColor": "#bbf7d0",
                  },
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  1004
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  38.45
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  2023-02-08
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  4211.152345617756
                </Text>
              </View>
              <View
                style={
                  [
                    {
                      "borderColor": "#000000",
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                    },
                    {
                      "justifyContent": "center",
                    },
                    undefined,
                    undefined,
                    undefined,
                    {
                      "flex": 1,
                    },
                    undefined,
                  ]
                }
              >
                <Text
                  style={
                    [
                      {
                        "fontSize": 12,
                        "textAlign": "center",
                      },
                      {
                        "backgroundColor": "transparent",
                      },
                    ]
                  }
                >
                  5
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
